### PR TITLE
Update configuration tooling

### PR DESCRIPTION
**Pull request by findepi**
https://github.com/airlift/airlift/pull/756 
\+ https://github.com/airlift/airlift/pull/757
\+ https://github.com/airlift/airlift/pull/758
\- refactors

### CODE REVIEW

File: configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java [hunk 1]
```diff
@@ -313,6 +354,26 @@ private AttributeMetadata buildAttributeMetadata(Class<T> configClass, Method co
         return builder.build();
     }
 
+    private static String getEffectivelyPropertyName(Method configMethod)
```
Comment (willmo83) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java]:
```suggestion
    private static String getEffectivePropertyName(Method configMethod)
```

File: configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java [hunk 2]
```diff
@@ -1538,6 +1548,23 @@ public void setIntValue(int value)
...
+    {
+        private String value;
+
+        public String getValue()
+        {
+            return value;
+        }
+
+        @Config("value")
+        @SlatedForRemoval(notEarlierThan = "Jan 2018")
```
Comment (willmo83) [inline configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java]: This name makes sense here, but `getNotEarlierThan()` sounds odd. Maybe just call this `removalDate`.

File: configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java (hunk 1 above)
Comment (willmo83) [inline configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java]:
```suggestion
    private static String getEffectivePropertyName(Method configMethod)
```

File: configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (findepi) [reply configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java]: my thinking is that actual final removal is a manual process, so i dont _guarantee_ a specific removal date. I only guarantee it wont be happen before certain point in time.

it  could be also `notUntil`, or just `after`... `after` implies "Feb+" while `notEarlierThan` means "Jan+"

`notUntil`?

[the remark this answers is not in the document]
Comment (willmo83) [reply configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java]: How about `expectedAfter` or `expectedBy`? `notUntil` sounds ok too.

[the remark this answers is not in the document]
Comment (findepi) [reply configuration/src/test/java/io/airlift/configuration/ConfigurationMetadataTest.java]: I'm settling on `notUntil`.

File: configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java (hunk 1 above)
Comment (findepi) [reply configuration/src/main/java/io/airlift/configuration/ConfigurationMetadata.java]: You're a mindful reader, @willmo83. Thanks for catching me.

**Comment by findepi** [comment]
I enabled Travis here and I added some commits that fix the build there.

**Pull request merged**